designs/2023-rule-performance-dashboard/README.md

```diff
@@ -0,0 +1,276 @@
+- Repo: eslint/eslint
+- Start Date: 2023-03-10
+- RFC PR: <https://github.com/eslint/rfcs/pull/108>
+- Authors: Mara Nikola Kiefer (@mnkiefer)
+
+# Performance Statistics
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+This document describes a new ESLint option `stats`, which enables the user to obtain a series of runtime *statistics* on top of their final lint results.
+
+These statistics include more granular information, such as the [parse](#timing-parse)-, [fix](#timing-fix)-, and [lint-times](#timing-lint) and the number of [fix passes](#fix-passes), [directives and violations](#directives-and-violations) encountered.
+
+| Stats |  | Description |
+|--------|-------|---|
+| **timing** | total  | The overall lint time that is spent on a file. This includes `parse-`, `fix-` and all `rules-` (total) times. |
+|        | parse | The time that is spent when parsing a file, that is when [`parse()`](#timing-parse) or `parseForEslint()` is called. |
+|        | rules | This is similar to the rule performance TIMING, except that it is more granular, that is, persisted as *per file per rule*.  |
+| **passTiming** | | An array of *timing* objects for each of the [fix passes](#fix-passes). In total, these amount to the (total) *timing* object for the respective file. |
```

Comment:
What about naming this `timingPasses`? That way, it will be grouped next to `timing` when the properties are listed alphabetically. Using common prefixes in property names is helpful for grouping and helpful for users to understand related properties.

Reply:
I followed @nzakas [suggestion for the naming](https://github.com/eslint/rfcs/pull/108#discussion_r1189103425). Please clarify with him first, I'm open to renaming it.

Reply:
I don't think `timingPasses` explains what the value actually is, nor do I think it necessarily needs to be next to `timing` alphabetically -- that seems like a pretty severe edge case to me.

Reply:
To expand a bit (because I got mixed up): I'd like to use the same words for the same types of values. `fixPasses` is a number so I'd expect something called `timingPasses` to also be a number. Right now `timing` is an object and `passTiming` is an object (albeit an array), so I think it makes sense, although might arguably be better as `passTimings` to indicate that it's an array containing multiple timing objects.

Reply:
Sounds reasonable. Just wanted to make sure we were being intentional and consistent with whatever preferred naming scheme. Naming numbers with the same format is a good example of that.